// @onramper/moonpay-adapter/build/KYC/items.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cryptocurrencyAddressTag = exports.creditCardTokenId = exports.transactionId = exports.cryptocurrencyAddress = exports.bankAccountNumberItem = exports.bankSortCodeItem = exports.bankIbanItem = exports.verifyPhoneCodeItem = exports.phoneCountryCodeItem = exports.phoneNumberItem = exports.verifyEmailCodeItem = exports.emailItem = exports.optionalStateItem = exports.stateItem = exports.countryItem = exports.postCodeItem = exports.townItem = exports.streetItem = exports.lastNameItem = exports.firstNameItem = exports.dateOfBirthItem = void 0;
exports.dateOfBirthItem = {
    type: "date",
    name: "dateOfBirth",
    humanName: "Date of Birth",
    data: [
        {
            type: "integer",
            humanName: "Day",
            name: "day",
        },
        {
            type: "integer",
            humanName: "Month",
            name: "month",
        },
        {
            type: "integer",
            humanName: "Year",
            name: "year",
        },
    ],
};
exports.firstNameItem = {
    type: "string",
    name: "firstName",
    humanName: "First name",
};
exports.lastNameItem = {
    type: "string",
    name: "lastName",
    humanName: "Last Name",
};
exports.streetItem = {
    type: "string",
    name: "street",
    humanName: "Street",
};
exports.townItem = {
    type: "string",
    name: "town",
    humanName: "City",
};
exports.postCodeItem = {
    type: "string",
    name: "postCode",
    humanName: "Postal Code",
};
exports.countryItem = {
    type: "string",
    name: "country",
    humanName: "Country",
};
exports.stateItem = {
    type: "string",
    name: "state",
    humanName: "State",
};
exports.optionalStateItem = Object.assign(Object.assign({}, exports.stateItem), { hint: "Only required if you ", required: false });
exports.emailItem = {
    type: "string",
    name: "email",
    humanName: "Email",
    hint: "We will send a code to your email.",
};
exports.verifyEmailCodeItem = {
    type: "string",
    name: "verifyEmailCode",
    humanName: "Email verification code",
};
exports.phoneNumberItem = {
    type: "integer",
    name: "phoneNumber",
    humanName: "Phone number",
};
exports.phoneCountryCodeItem = {
    type: "integer",
    name: "phoneCountryCode",
    humanName: "Country code",
};
exports.verifyPhoneCodeItem = {
    type: "string",
    name: "verifyPhoneCode",
    humanName: "Phone verification code",
};
exports.bankIbanItem = {
    type: "string",
    name: "bankIban",
    humanName: "IBAN",
};
exports.bankSortCodeItem = {
    type: "string",
    name: "bankSortCode",
    humanName: "Sort code",
};
exports.bankAccountNumberItem = {
    type: "string",
    name: "bankAccountNumber",
    humanName: "Bank account number",
};
exports.cryptocurrencyAddress = {
    type: "string",
    name: "cryptocurrencyAddress",
    humanName: "Cryptocurrency wallet address",
};
exports.transactionId = {
    type: "string",
    name: "transactionId",
    humanName: "Transaction Id",
};
exports.creditCardTokenId = {
    type: "string",
    name: "ccTokenId",
    humanName: "Credit Card Token Id",
};
exports.cryptocurrencyAddressTag = {
    type: "string",
    name: "cryptocurrencyAddressTag",
    humanName: "Cryptocurrency address tag",
    required: false,
};
